### PR TITLE
Build: Upgrade `esbuild` (`0.20.1`)

### LINKED ISSUE #26112: [Bug]: Cannot start service: Host version "0.18.20" does not match binary version "0.19.7"
### Describe the bug

I get this error when I start storybook:
```
> storybook dev -p 6006

@storybook/cli v7.6.17

X [ERROR] Cannot start service: Host version "0.18.20" does not match binary version "0.19.7"

1 error
Error: The service was stopped
    at .\node_modules\.pnpm\registry.npmjs.org+esbuild@0.18.20\node_modules\esbuild\lib\main.js:826:29
    at responseCallbacks.<computed> (.\node_modules\.pnpm\registry.npmjs.org+esbuild@0.18.20\node_modules\esbuild\lib\main.js:703:9)
    at Socket.afterClose (.\node_modules\.pnpm\registry.npmjs.org+esbuild@0.18.20\node_modules\esbuild\lib\main.js:693:28)
    at Socket.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1376:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.
```

When I did install latest Storybook `7.6.17` I noticed this:
```
 WARN  Issues with peer dependencies found
.
└─┬ @storybook/react-vite 7.6.17
  └─┬ @vitejs/plugin-react 3.1.0
    └── ✕ unmet peer vite@^4.1.0-beta.0: found 5.1.3
```

Changed my vite version to `4.1.0-beta.0` now storybook works again.

But the thing is that I have been using vite 5 since November without any problems.

### To Reproduce

Install all latest versions and vite 5. This is my package.json:

```
{
	"name": "@X/components",
	"version": "0.7.12",
	"type": "module",
	"main": "dist/main.js",
	"types": "dist/main.d.ts",
	"files": [
		"dist"
	],
	"scripts": {
		"preinstall": "npx only-allow pnpm",
		"prepublishOnly": "pnpm run build-sb",
		"sb": "storybook dev -p 6006",
		"build-sb": "vite build --config vitePublish.config.ts",
		"chromatic": "npx chromatic --auto-accept-changes --project-token=xxxxxxxxxxxxxx",
		"build-storybook": "sb build",
		"refreshVSToken": "npx vsts-npm-auth -config .npmrc"
	},
	"peerDependencies": {
		"@nextui-org/react": "^2.2.9",
		"framer-motion": "^11.0.5",
		"react": "^18.2.0",
		"react-dom": "^18.2.0",
		"tailwind-variants": "^0.2.0",
		"tailwindcss": "^3.4.1",
		"clsx": "^2.1.0",
		"tailwind-merge": "^2.2.1"
	},
	"devDependencies": {
		"@nextui-org/react": "^2.2.9",
		"@storybook/addon-a11y": "^7.6.17",
		"@storybook/addon-actions": "^7.6.17",
		"@storybook/addon-essentials": "^7.6.17",
		"@storybook/addon-interactions": "^7.6.17",
		"@storybook/addon-links": "^7.6.17",
		"@storybook/addon-onboarding": "^1.0.11",
		"@storybook/addon-styling": "^1.3.7",
		"@storybook/blocks": "^7.6.17",
		"@storybook/manager-api": "^7.6.17",
		"@storybook/react": "^7.6.17",
		"@storybook/react-vite": "^7.6.17",
		"@storybook/theming": "^7.6.17",
		"@types/node": "20.11.19",
		"@types/react": "18.2.57",
		"@types/react-dom": "18.2.19",
		"@typescript-eslint/eslint-plugin": "^7.0.2",
		"@typescript-eslint/parser": "^7.0.2",
		"@vitejs/plugin-react": "^4.2.1",
		"autoprefixer": "^10.4.17",
		"chromatic": "^10.9.6",
		"eslint": "8.56.0",
		"eslint-config-prettier": "^9.1.0",
		"eslint-plugin-react-hooks": "^4.6.0",
		"eslint-plugin-react-refresh": "^0.4.5",
		"eslint-plugin-storybook": "^0.8.0",
		"framer-motion": "^11.0.5",
		"glob": "^10.3.10",
		"postcss": "^8.4.35",
		"prettier": "3.2.5",
		"prettier-plugin-tailwindcss": "^0.5.11",
		"react": "^18.2.0",
		"react-dom": "^18.2.0",
		"rollup-plugin-preserve-directives": "^0.4.0",
		"storybook": "^7.6.17",
		"tailwind-variants": "^0.2.0",
		"tailwindcss": "^3.4.1",
		"typescript": "^5.3.3",
		"vite": "^4.5.2",
		"vite-plugin-dts": "^3.7.2",
		"vite-plugin-lib-inject-css": "^1.3.0",
		"clsx": "^2.1.0",
		"tailwind-merge": "^2.2.1"
	}
}

```

### System

```bash
Storybook Environment Info:

  System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz   
  Binaries:
    Node: 20.7.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.1.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.8.0 - ~\AppData\Local\pnpm\pnpm.EXE <----- active
  Browsers:
    Edge: Chromium (121.0.2277.128)
  npmPackages:
    @storybook/addon-a11y: ^7.6.17 => 7.6.17
    @storybook/addon-actions: ^7.6.17 => 7.6.17
    @storybook/addon-essentials: ^7.6.17 => 7.6.17
    @storybook/addon-interactions: ^7.6.17 => 7.6.17
    @storybook/addon-links: ^7.6.17 => 7.6.17
    @storybook/addon-onboarding: ^1.0.11 => 1.0.11
    @storybook/addon-styling: ^1.3.7 => 1.3.7
    @storybook/blocks: ^7.6.17 => 7.6.17
    @storybook/manager-api: ^7.6.17 => 7.6.17
    @storybook/react: ^7.6.17 => 7.6.17
    @storybook/react-vite: ^7.6.17 => 7.6.17
    @storybook/theming: ^7.6.17 => 7.6.17
    chromatic: ^10.9.6 => 10.9.6
    eslint-plugin-storybook: ^0.8.0 => 0.8.0
    storybook: ^7.6.17 => 7.6.17
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 43081j**
This just upgrades esbuild to 0.20.1

Do you have any concerns with doing this?

The reason (other than staying up to date) is that many other peers are now on 0.20.x, so storybook's various packages are resulting in our dependency tree have many copies of esbuild instead of a single de-duped copy.

### CODE REVIEW

File: code/builders/builder-manager/package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "@yarnpkg/esbuild-plugin-pnp": "^3.0.0-rc.10",
     "browser-assert": "^1.2.1",
     "ejs": "^3.1.8",
-    "esbuild": "^0.18.0",
```
Comment (yannbf) [inline code/builders/builder-manager/package.json:55]: @valentinpalkovic do you remember why we are on 0.18? I reckon something happened in the past but I just don't remember

Comment (43081j) [reply code/builders/builder-manager/package.json:55]: whats the best way to prove everything is fine other than CI being happy?

if you can point me in the right direction, i can try do further testing

**Review by ndelangen — APPROVED** [review]

File: code/builders/builder-manager/package.json (hunk 1 above)
Comment (ndelangen) [reply code/builders/builder-manager/package.json:55]: Hmm,.. maybe this:
https://github.com/storybookjs/storybook/issues/26112

File: code/lib/core-common/package.json [hunk 2]
```diff
@@ -52,7 +52,7 @@
     "@yarnpkg/libzip": "2.3.0",
     "chalk": "^4.1.0",
     "cross-spawn": "^7.0.3",
-    "esbuild": "^0.18.0",
+    "esbuild": "^18.0.0 || ^19.0.0 || ^0.20.0",
```
Comment (levrik) [inline code/lib/core-common/package.json:55]: Typo?

`18.0.0` -> `0.18.0`
`19.0.0` -> `0.19.0`

Comment (43081j) [reply code/lib/core-common/package.json:55]: it is a typo! 👀  this was added by @ndelangen i think, probably trying to account for mixed versions in dependency trees. should be easy enough to fix at least

**Pull request merged**